### items to check
- number of articles
- number of authors
- check under-splitting: different author share the same id -- id labeling mistake
- check over-splitting: the same author have different ids

In [1]:
import pandas as pd
import os

In [2]:
os.getcwd()

'C:\\Users\\mansj\\IQL\\ExRx Data Quality'

In [3]:
author_data = pd.read_csv(r"C:/Users/mansj/IQL/data/exrx_author.csv", encoding="UTF-8")

In [4]:
author_data.head()

ID                                              Title author_given_name  \
0   1  Physical activity in depressed elderly. a syst...             Gioia   
1   1  Physical activity in depressed elderly. a syst...    Mauro Giovanni   
2   2                            Exercise for depression          Gary. M.   
3   2                            Exercise for depression             Kerry   
4   2                            Exercise for depression        Carolyn A.   

  author_surname    author_id  
0           Mura   7005321937  
1          Carta   7006714673  
2         Cooney  26028761700  
3           Dwan  25027473800  
4          Greig   7003441733

In [5]:
# total unique articles 441
len(author_data.ID.unique())

441

In [6]:
# total unqiye author id's
len(author_data.author_id.unique())

1961

## Check under-splitting

In [15]:
full_name = author_data.author_given_name + " " + author_data.author_surname
name_df = pd.DataFrame(data = {'full_name':author_data.author_given_name + " " + author_data.author_surname,
                               'author_id':author_data.author_id})
name_df


full_name    author_id
0               Gioia Mura   7005321937
1     Mauro Giovanni Carta   7006714673
2          Gary. M. Cooney  26028761700
3               Kerry Dwan  25027473800
4         Carolyn A. Greig   7003441733
...                    ...          ...
2501            MH Trivedi   7103153350
2502               AL Dunn   7201720357
2503             DI Galper   6603479223
2504            JB Kampert   6701722656
2505              TL Green          189

[2506 rows x 2 columns]

In [8]:
len(name_df.drop_duplicates())

2079

In [11]:
check_u_split = name_df.groupby("author_id").count()
check_u_split[check_u_split["full_name"]>1].rename(columns = {"full_name":"no_of_names"}).reset_index().merge(name_df, 
                                                                                                              on="author_id").drop_duplicates().to_csv("check_split.csv")

### No undersplitting found

## Check under-splitting

In [24]:
auth_name_id = name_df.drop_duplicates().groupby("full_name").count().sort_values(by='author_id', ascending = False)
auth_name_id[auth_name_id["author_id"]>1].rename(columns={"author_id" : "no_of_ids"}).reset_index().merge(name_df, on = "full_name")

full_name  no_of_ids    author_id
0           Suhui Ma          2           46
1           Suhui Ma          2  49863886900
2           Suhui Ma          2           46
3     Anders Hovland          2          190
4     Anders Hovland          2          199
5    Changxiang Chen          2  53864637600
6    Changxiang Chen          2           45
7   Tamara B. Harris          2  57220774479
8   Tamara B. Harris          2  35375662400
9           Yan Chen          2           18
10          Yan Chen          2           20
11            Na Dou          2           89
12            Na Dou          2  49863133400

Corrected the under-splitting manually by replacing the author_id by the largest author_id present for the authors

    Suhui Ma : 49863886900
    Anders Hovland : 199
    Changxiang Chen : 53864637600
    Tamara B. Harris : 57220774479
    Yan Chen : 20
    Na Dou : 49863133400

### Final check after corrections

In [28]:
author_data = pd.read_csv(r"C:/Users/mansj/IQL/data/exrx_author_final.csv", encoding="UTF-8")
name_df = pd.DataFrame(data = {'full_name':author_data.author_given_name + " " + author_data.author_surname,
                               'author_id':author_data.author_id})
auth_name_id = name_df.drop_duplicates().groupby("full_name").count().sort_values(by='author_id', ascending = False)
auth_name_id[auth_name_id["author_id"]>1].rename(columns={"author_id" : "no_of_ids"}).reset_index().merge(name_df, on = "full_name")

Empty DataFrame
Columns: [no_of_ids, full_name, author_id]
Index: []